src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Extensions/DesignTimeDirectiveTargetExtension.cs

```diff
@@ -152,6 +152,17 @@ private void WriteDesignTimeDirectiveToken(CodeRenderingContext context, DesignT
 
                 case DirectiveTokenKind.String:
 
+                    var stringSyntax = context.DocumentKind switch
+                    {
+                        "mvc.1.0.razor-page" => "Route",
```

Comment:
It does mean that we can't use the nice switch expression syntax, but I'd prefer to use the static readonly `RazorPageDocumentClassifierPass.RazorPageDocumentKind` and `ComponentDocumentClassifierPass.ComponentDocumentKind` fields instead. Sorry.

Reply:
@333fred Your suggestion isn't possible. What do you want me to do here?

Reply:
That's unfortunate. I guess we'll leave it as is.